test/function/samples/preserve-modules-virtual-modules/_config.js

```diff
@@ -0,0 +1,36 @@
+const assert = require('assert');
+
+module.exports = {
+	description: 'Generates actual files for virtual modules when preserving modules',
+	options: {
+		input: ['main.js'],
+		experimentalCodeSplitting: true,
```

Comment:
arent those like conflicting settings? preserving modules & code splitting?

Reply:
Not in rollup < 1.0. `experimentalCodeSplitting` internally uses the "new" code-splitting pipeline which is a prerequisite for preserving modules, which is like the ultimate form of code-splitting. In rollup 1.0 of course, this is the only pipeline that exists.